demo/popcha-client/web/src/main.py

```diff
@@ -0,0 +1,222 @@
+"""
+This files includes the basis of the example server. Here are managed all
+HTTP/S requests.
+"""
+import json
+from os import path
+import secrets
+import urllib.parse
+from typing import IO
+
+from flask import Flask, redirect, request, Response, render_template
+from flask_wtf import CSRFProtect
+
+from src.counterApp import CounterApp
+from src.authentication import Authentication
+
+# Define the global variables
+config: dict = {}
+core_app: CounterApp
+authenticationProvider: Authentication
+app = Flask("Example_authentication_server",
+            template_folder = path.join(path.dirname(__file__), "templates")
+            )
+
+
+def check_config(config_file: IO) -> bool:
+    """
+    Checks that the global config file is valid. Updates the updatable
+    parameters in the file.
+    :param config_file:  the file where the configuration is written
+    :return: True if the configurations is valid
+    """
+    if "client_id" not in config.keys() or config["client_id"] == '':
+        config["client_id"] = secrets.token_urlsafe(32)
+        config_file.seek(0)
+        config_file.write(json.dumps(config))
+        return False
+    if "host_url" not in config.keys() or config["host_url"] == "":
+        raise ValueError(
+                "The \"server_url\" property should be set in "
+                "config.json and not empty"
+                )
+    if "host_port" not in config.keys() or config["host_port"] < 1:
+        raise ValueError(
+                "The \"server_port\" should be set in config.json "
+                "and greater than 0"
+                )
+    return True
+
+
+def check_provider(provider: dict) -> bool:
+    """
+    Check that a single provider is valid. It only performs static checks and
+    no connectivity tests.
+    :param provider: The provider to check
+    :return: True if the provider is valid
+    """
+    valid_lao_id = (("lao_id" in provider) and (len(provider["lao_id"]) == 44)
+                    and provider["lao_id"].endswith("="))
+    valid_domain = (("domain" in provider) and ('/' not in provider["domain"])
+                    and not provider["domain"].startswith("http"))
+    valid_public_key = len(provider["public_key"]) > 0
+    return valid_lao_id and valid_domain and valid_public_key
```

Comment:
The formatting here is not very readable 😅 Also for some "magic" numbers (44) it would help adding a short comment explaining what the check does.